src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -1364,8 +1378,372 @@ def get_unit_spike_train(
         return frames[start_index:end_index].astype("int64", copy=False)
 
 
+def _find_timeseries_from_backend(group, path="", result=None, backend="hdf5"):
+    """
+    Recursively searches for groups with TimeSeries neurodata_type in hdf5 or zarr object,
+    and returns a list with their paths.
+    """
+    if backend == "hdf5":
+        import h5py
+
+        group_class = h5py.Group
+    else:
+        import zarr
+
+        group_class = zarr.Group
+
+    if result is None:
+        result = []
+
+    for name, value in group.items():
+        if isinstance(value, group_class):
+            current_path = f"{path}/{name}" if path else name
+            if value.attrs.get("neurodata_type") == "TimeSeries":
+                result.append(current_path)
+            _find_timeseries_from_backend(value, current_path, result, backend)
+    return result
+
+
+class NwbTimeSeriesExtractor(BaseRecording, _BaseNWBExtractor):
+    """Load a TimeSeries from an NWBFile as a RecordingExtractor.
+
+    Parameters
+    ----------
+    file_path : str | Path | None
+        Path to NWB file or an s3 URL. Use this parameter to specify the file location
+        if not using the `file` parameter.
+    timeseries_path : str | None
+        The path to the TimeSeries object within the NWB file. This parameter is required
+        when the NWB file contains multiple TimeSeries objects. The path corresponds to
+        the location within the NWB file hierarchy, e.g. 'acquisition/MyTimeSeries'.
+    load_time_vector : bool, default: False
+        If True, the time vector is loaded into the recording object. Useful when
+        precise timing information is needed.
+    samples_for_rate_estimation : int, default: 1000
+        The number of timestamp samples used for estimating the sampling rate when
```

Comment:
```suggestion
        The number of timestamps used for estimating the sampling rate when
```
I think this is clear :)